plugins/autodim.koplugin/main.lua

```diff
@@ -344,7 +324,7 @@ function AutoDim:ramp_task()
         self.ramp_event_countdown = self.ramp_event_countdown - 1
         if self.ramp_event_countdown <= 0 then
             -- Update footer on every self.ramp_event_countdown call
-            self:updateFooter()
+            UIManager:broadcastEvent("UpdateFooter")
```

Comment:
Just stumble on this line while doing `rgrep broadcast.*ooter frontend plugins`.
I don't think it works if we provide just a string. At least, it's the single bit of code that does that.
The correct way is as it was in the remove chunk above:
`UIManager:broadcastEvent(Event:new("UpdateFooter"...)`.

Reply:
Nice catch ;).

In fact, the only thing that prevents it from crashing is the fact that strings are a metatable, so you *can* try to access a field in a string (yay, Lua), but that's an implementation detail ;o).

On the upside, I think not doing anything is the right call: refreshes would race with the frontlight changes and ruin the effect. Only updating on end makes the most sense.

Reply:
I'm assuming the ramping is fast for that last statement, if that isn't actually the case, then, yeah, update the footer ;).

(Ping @zwim)